datasquare.go

```diff
@@ -307,16 +307,13 @@ func (ds *dataSquare) SetCell(x uint, y uint, newChunk []byte) error {
 	return nil
 }
 
-// setCell sets a specific cell. setCell will overwrite any existing value.
-// Returns an error if the newChunk is not the correct size.
-func (ds *dataSquare) setCell(x uint, y uint, newChunk []byte) error {
-	if len(newChunk) != int(ds.chunkSize) {
-		return fmt.Errorf("cannot set cell with chunk size %d because dataSquare chunk size is %d", len(newChunk), ds.chunkSize)
-	}
+// setCell overwrites the contents of a specific cell. setCell does not perform
+// any input validation so most usecases should use `SetCell` instead of
+// `setCell`. This method exists strictly for testing.
+func (ds *dataSquare) setCell(x uint, y uint, newChunk []byte) {
```

Comment:
Since this is meant to be used for testing purposes, I think it's a good idea to move it to a test package. That way, we can avoid any accidental misuse.

Reply:
One alternative  solution could be to call `SetCell` in the test only for non-nill values, i.e., in [here](https://github.com/celestiaorg/rsmt2d/blob/25576208f90d3d171e5cbf7407704c1d61ae6bf1/extendeddatacrossword_test.go#L209).
```go
  for i, coords := range test.coords {
          x := coords[0]
          y := coords[1]
          if test.values[i]!= nil {
             eds.setCell(x, y, test.values[i])
          }
  }
```

Reply:
Hmm I don't understand the proposed alternative solution. The test wants to overwrite nil and non-nil values. Do you mind rephrasing? Also are you referring to `setCell` or `SetCell`?

Reply:
> I think it's a good idea to move it to a test package

Can do!

Reply:
> Hmm I don't understand the proposed alternative solution. The test wants to overwrite nil and non-nil values. Do you mind rephrasing? Also are you referring to `setCell` or `SetCell`?

Oh, I mistakenly assumed that the `eds` contains `nil` cells by default so the tests are can be reduced to altering nil cells to non-nil cells, but now I see that `eds` initially gets populated with non-nil values in `createTestEds`, so my alternative solution does not apply in here.